lib/wcc/contentful/model_methods.rb

```diff
@@ -51,31 +95,45 @@ def _resolve_field(field_name, depth = 1, context = {})
     var_name = '@' + field_name
     return unless val = instance_variable_get(var_name)
 
-    context[id] ||= self
+    context = sys.context.to_h.merge(context)
     load =
       ->(raw) {
         id = raw.dig('sys', 'id')
-        return context[id] if context.key?(id)
-        m = context[id] =
-              if raw.dig('sys', 'type') == 'Link'
-                WCC::Contentful::Model.find(id)
-              else
-                content_type = content_type_from_raw(raw)
-                const = WCC::Contentful::Model.resolve_constant(content_type)
-                const.new(raw, context)
-              end
-
-        m.resolve(depth: depth - 1, context: context) if m && depth > 1
+        already_resolved = context[:backlinks]&.find { |m| m.id == id }
```

Comment:
does this handle the case where we might ask for greater depth after previously resolving a link?

Reply:
It does not - I'll add that test case